machine_learning_hep/analysis/analyzer_jet.py

```diff
@@ -3765,6 +4366,64 @@ def get_simulated_yields(self, file_path: str, dim: int, prompt: bool):
         df_sim = adjust_nsd(df_sim)
         # Adjust z values.
         df_sim = adjust_z(df_sim)
+        # pt-dependent rapidity cut
+        sel_cand_array = selectfidacc(df_sim["pt_cand"].values, df_sim["y_cand"].values)
+        df_sim = df_sim[np.array(sel_cand_array, dtype=bool)]
+        # prompt vs. non-prompt selection
+        pdg_parton_good = 4 if prompt else 5
+        df_sim = df_sim[df_sim["pdg_parton"] == pdg_parton_good]
+        if xsec:
+            file_fonll = TFile.Open(self.filename_fonll)
+            if "D0" in self.case:
+                fonll_pred = file_fonll.Get("hD0KpifromBpred_central_corr")
+            elif "Lc" in self.case:
+                fonll_pred = file_fonll.Get("hLcK0sppfromBpred_central_corr")
+            fonll_hist = fonll_pred.Clone("fonll_hist")
+            x_max = fonll_hist.GetXaxis().GetXmax()
+            x_min = fonll_hist.GetXaxis().GetXmin()
+            nbins = fonll_hist.GetNbinsX()
+            rebin_const = int(nbins/(x_max-x_min))
```

Comment:
need to check that rebin_const is an integer by default and only rebin if its bigger > 1